File: Model/Nodes/UnaryMinus.vb
Imports System.Collections.Generic

Namespace Isocline.Nodes
    Friend Class UnaryMinus
        Implements INode
        Private valueField As INode

        Public Sub New(ByVal value As INode)
            valueField = value
        End Sub

        Public Function Value(ByVal variables As Dictionary(Of String, Double)) As Double Implements INode.Value
            Return -valueField.Value(variables)
        End Function
    End Class
End Namespace

File: Model/Nodes/UnaryMinus.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Isocline.Nodes
{
    class UnaryMinus: INode
    {
        private INode value;

        public UnaryMinus(INode value)
        {
            this.value = value;
        }

        public double Value(Dictionary<string, double> variables)
        {
            return - value.Value(variables);
        }
    }
}
